### PR TITLE
45823 - Appointment List - Mobile View v2

### PR DESCRIPTION
**Pull request by vbahinwillit**
## Summary
Mobile view for the VAOS Homepage appointment list redesign

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#45823

## Testing done
Unit testing and e2e testing

## What areas of the site does it impact?
VAOS

## Tasks
- [x] #22899 
- [x] #22890 
- [x] #23008 
- [x] #23059 
- [x] #23007 

## Acceptance criteria
- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by jenniemc — CHANGES_REQUESTED** [review]
For this iteration, I only reviewed the files supporting `UpcomingAppointmentLayout` component. So far it's looking real good in the desktop and mobile small-screen version. I included some suggestions for accessibility.

You may want to check out the text alignment on the print version.
Screenshot 2023-01-11 at 3 08 13 PM

**Review by rmessina1010 — APPROVED** [review]
LGTM

File: src/applications/vaos/appointment-list/models/AppointmentModel.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  getTimeZoneAbbr() {
+    const { abbreviation } = this.getAppointmentTimezone();
+    return abbreviation;
+  }
+
+  getAppointmentTimezone() {
+    return getAppointmentTimezone(this.data);
+  }
+
+  getAppointmentDetails() {
```
[the remark this answers is not in the document]
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/models/AppointmentModel.js:143]: Thanks for the suggestiong. Changed to 'getAppointmentLocality'.

File: src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentColumnLayout.jsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                {model.getStartDate().format('D')}
+              </h3>
+            </AppointmentColumn>
+            <AppointmentColumn
+              className="xsmall-screen:vads-u-order--first small-screen:vads-u-margin-left--1"
+              size="1"
+              style={{ minWidth: '25px', maxWidth: '25px' }}
+            >
+              <span>{model.getStartDate().format('ddd')}</span>
+              <span className="sr-only"> {model.getTimeZoneAbbr()}</span>
```
[the remark this answers is not in the document]
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentColumnLayout.jsx:41]: Thanks for the suggestion. I will look into what Voice Over announces when we start 508 testing. The announcement of timezone was in the original code though.

**Review by jenniemc — APPROVED** [review]
LGTM!!

**Pull request merged**